### ISSUE TITLE
Rule proposal: no-ineffective-disable-eslint

### ISSUE THREAD

**Issue by gajus** [eslint/eslint#5202]
This rule would report when when `eslint-disable` comment is used without an effect, e.g.

```js
/* eslint-disable no-console */
var test;
```

In this code snippet `no-console` has no effect.

The need for this rule is to prevent meaningless comments spread throughout the codebase.

**Comment by eslintbot** [comment]
@gajus Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by ilyavolodin** [comment]
Sorry, I don't think something like that is possible. In order to figure out if rule has any errors/warnings, we need to run it, we can't run rules from inside other rules. More than that, rules that are disabled by inline comments do not even execute, so we can't even check if they error or not. Overall, I don't think something like that is feasible even in the future.

**Comment by gajus** [comment]
Wouldn't it be as simple as removing the ESLint comments, converting code to AST and feeding it to the rule, then finally checking all the errors that have been triggered and matching their lines against the original comment position?

**Comment by gajus** [comment]
Yes, that would basically require rule itself to use CLIEngine to validate the code using all other inherited rules. (or at least those mentioned in the comments)

**Comment by ilyavolodin** [comment]
Yes, that's what I was getting to. You have to have rule access CLIEngine to do something like that. We can't do that, that would be a horrible architectural decision. Our rules currently are completely stand-alone and have absolutely no knowledge about anything above them, other then `RuleContext`. Breaking this patter seems like asking for trouble in the future.

**Comment by mk-pmb** [comment]
Could a rule add a task to be run later, when the results of first linting are in? Kinda like multipass video encoding. The defered task could lint again w/o the disable comments and compare.
